### PR TITLE
interop: update op-node <> op-supervisor data exchange

### LINKED ISSUE #12077: Interop: op-supervisor local-safe data-ingestion from op-node
The op-node must feed into the op-supervisor, the op-supervisor does not rely on the EL for state.
The op-supervisor was pulling `safe` and `finalized` from the EL
(this originated from prototype code), intending to retrieve `local-safe`, but it was the `cross-safe`.
Creating a cyclic dependency, keeping `cross-safe` stuck at genesis.

To fix this, we need to modify the `op-node/rollup/interop` deriver to emit RPC updates to the op-supervisor when local-safe data changes.

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

Work in progress

**Tests**

Work in progress

**Additional context**

**Metadata**

Fix https://github.com/ethereum-optimism/optimism/issues/12077

### CODE REVIEW

File: op-node/rollup/engine/events.go [hunk 1]
```diff
@@ -419,8 +435,11 @@ func (d *EngDeriver) OnEvent(ev event.Event) bool {
 			return true
 		}
 		d.ec.SetFinalizedHead(x.Ref)
+		d.emitter.Emit(FinalizedUpdateEvent{Ref: x.Ref})
 		// Try to apply the forkchoice changes
 		d.emitter.Emit(TryUpdateEngineEvent{})
+	case RequestFinalizedUpdateEvent:
+
```
Comment (axelKingsley) [inline op-node/rollup/engine/events.go]: is this meant to be blank, or is there intended to be some logic which emits a new FinalizedUpdateEvent that we need to put here?

File: op-node/rollup/interop/interop.go [hunk 2]
```diff
@@ -78,84 +82,168 @@ func (d *InteropDeriver) OnEvent(ev event.Event) bool {
...
-		blockSafety, err := d.backend.CheckBlock(ctx, d.chainID, candidate.Hash, candidate.Number)
-		if err != nil {
-			d.log.Warn("Failed to check interop safety of local-safe block", "err", err)
-			break
-		}
-		derivedFrom, ok := d.derivedFrom[candidate.Hash]
-		if !ok {
-			break
+		if err := d.onCrossSafeUpdateEvent(x); err != nil {
+			d.log.Error("Failed to process cross-safe update", "err", err)
```
Comment (axelKingsley) [inline op-node/rollup/interop/interop.go:96]: TYVM for splitting these out

**Review by axelKingsley — COMMENTED** [review]
There's lots of overlapping changes from the branch this PR is based on. I am not sure why github hasn't figured that out, but it's a little difficult to see the changes through them.

**Comment by axelKingsley** [comment]
> There's lots of overlapping changes from the branch this PR is based on. I am not sure why github hasn't figured that out, but it's a little difficult to see the changes through them.

Lol, *as* I hit the button the "Refresh" icon appeared, looks like this aspect is resolved.

File: op-node/rollup/engine/events.go (hunk 1 above)
Comment (protolambda) [reply op-node/rollup/engine/events.go]: work in progress

Comment (protolambda) [reply op-node/rollup/engine/events.go]: fixed

**Comment by protolambda** [comment]
Closing this, it was covered in https://github.com/ethereum-optimism/optimism/pull/12204

**Pull request closed without merging**